src/main/java/org/gridsuite/merge/orchestrator/server/MergeOrchestratorService.java

```diff
@@ -124,26 +124,32 @@ public void consume(Message<String> message) {
                 }
 
                 if (!processConfigs.isEmpty()) {
-                    // import IGM into the network store
-                    UUID networkUuid = caseFetcherService.importCase(caseUuid);
-                    // check IGM quality
-                    boolean valid = igmQualityCheckService.check(networkUuid);
-
-                    merge(processConfigs.get(0), dateTime, date, tso, valid, networkUuid);
-
-                    for (ProcessConfig processConfig : processConfigs.subList(1, processConfigs.size())) {
-                        // import IGM into the network store
-                        UUID processConfigNetworkUuid = caseFetcherService.importCase(caseUuid);
-                        merge(processConfig, dateTime, date, tso, valid, processConfigNetworkUuid);
-                    }
+                    Mono<UUID> networkUuidMono = caseFetcherService.importCase(caseUuid);
+
+                    Mono<Boolean> validMono = networkUuidMono.flatMap(networkUuid ->
+                       // check IGM quality
+                        igmQualityCheckService.check(networkUuid)
+                    );
+
+                    validMono.zipWith(networkUuidMono, (valid, networkUuid) ->
```

Comment:
remove zipwith and nest in previous flatmap ?

Reply:
@jonenst Do you think that nesting ops are better than using zipWith? 
For my point of view i think it's better to avoid nesting  for better readability. then the  readiness of validMono will launch many operations in // (see line 139), we should avoid to wait until the finish of the first merge to launch those merges.

Reply:
> Do you think that nesting ops are better than using zipWith?

No general answer.. I see now why you used zipwith in this particular case. It's ok.

In a different (more simple) case, compare the following codes:
```
MonoC = MonoA.flatmap(A -> 
     MonoB = B(A);
     return MonoC = C(A,B);
 )
``` 
vs

```
MonoB = MonoA.flatmap(A -> 
     return MonoB = B(A);
 )
MonoMonoC = MonoA.zipWith(MonoB, (A,B) -> C(A,B))
```
I think the first one is better, that's why I recommended nesting. But in your case, MonoB is reused later and MonoMonoC is not used so the zipwith version is better.

Reply:
So you did a good literal translation from the original code from the blocking version to the reactor version. But I still find the code a bit weird (even before your changes). For example, in the case where the network fails the validity check, we still import it as many times as the number of processes. We should either do all the imports before checking the validity, or don't import if the network is invalid.

I have a feeling the code can be simplified, for example by avoiding to write the merge() code twice (once for the first of the list, once for the rest of the list).

Reply:
I call the merge method  twice to not re-import the  case for the first merge then avoid time waste.

Reply:
yes I understood the code, but I think you can achieve the same optimisation with more straightforward code where you only write merge() once in one loop

Reply:
Here's an example (untested) (and still importing all the cases even if not valid like before) (and maybe can still be improved ?):
```java
if (!processConfigs.isEmpty()) {
    List<Mono<UUID>> imported = processConfigs.stream().map( ignored -> caseFetcherService.importCase(caseUuid)).collect(Collectors.toList());
    
    // Use the first network to check IGM validity
    imported.get(0).flatMap(networkUuid ->
       igmQualityCheckService.check(networkUuid)
    ).flatMap(valid -> {
        // write each IGM status
        var l = Streams.zip(processConfigs.stream(), imported.stream(), (processConfig, processConfigNetworkUuid) -> 
            processConfigNetworkUuid.flatMap(uuid -> merge(processConfig, dateTime, date, tso, valid, uuid))).collect(Collectors.toList());
        return Flux.fromIterable(l).flatMap(Function.identity()).collectList();
    }).subscribe();
}
```